js/taxclock/tc-incomecalculator.js

```diff
@@ -89,15 +89,38 @@ var IncomeCalculator = function() {
   //let's make sure the day ends at 5pm by adding 60 minutes to the 480 WORKDAY_MINS.
   this.END_OF_DAY = this.START_OF_DAY.clone().add(this.WORKDAY_MINS + 60, 'minutes');
 
+  // Calculate NSSF contribution based on new rates (effective Feb 2025)
+  // https://assets.kpmg.com/content/dam/kpmg/ke/pdf/tax/2025/Enhanced%20NSSF%20rates%20effective%20February%202025%20-%20Tax%20Alert.pdf
+  this.calculateNSSF = function(income) {
+    // Tier I: Applies to the first Ksh 8,000 at 6%
+    var tier1 = Math.min(income, 8000);
+    // Tier II: Applies to income between Ksh 8,001 and Ksh 72,000 (max Ksh 64,000) at 6%
+    var tier2 = Math.min(Math.max(income - 8000, 0), 64000);
+    var nssfTier1 = tier1 * 0.06;
+    var nssfTier2 = tier2 * 0.06;
+    return nssfTier1 + nssfTier2;
+  };
+
   this.calculateIncomeBreakdown = function(income) {
     var info = {};
 
     info.income = income;
 
-    // income tax
-    info.incomeTax = self.incomeTax(info);
-    // after tax income
-    info.netIncome = income - info.incomeTax;
+    // Calculate statutory contributions/deductions
+    info.nssf = self.calculateNSSF(income);
+    info.shif = (income * this.SHIF) / 100;                  // 2.75% Social Health Insurance Fund (SHIF)
+    info.housingLevy = (income * this.HOUSING_LEVY) / 100;   // 1.5% Affordable Housing Levy
+
+    info.statutory_deductions = info.nssf + info.shif + info.housingLevy;
```

Comment:
If we're deducting nssf, etc., should the information on the site be updated?
>What is your monthly income (before tax**)?
>** This should be your income before tax but _after deductions such as pension and health insurance._
>

<img width="556" alt="Screenshot 2025-03-12 at 18 25 45" src="https://github.com/user-attachments/assets/6201544e-ffef-4006-bf42-e1bf539d5ee0" />

Reply:
@kilemensi you are right. That must have slipped through my mind. However, my proposition to include it was mainly informed by a search of all the other applicable taxes in Kenya besides the personal income tax; https://taxsummaries.pwc.com/kenya/individual/other-taxes. Would be glad to hear your thoughts on this.

Reply:
I'm cool with us including it @elric-wamugu if it'll make more sense to Kenyas. We just need to update the comment above to indicate this change.

Reply:
Hey @kilemensi I've updated the message in this [PR #9](https://github.com/CodeForAfrica/TaxClock.Kenya/pull/9)